## SciPro ID: 2026SS_07

# Homework_2: The Average Speed of a Falling Object

***

### Task 1

The goal of this task is to create two user-defined functions. 

The first function calculates the distance an object falls under the standard acceleration of gravity while ignoring air resistance.

For this we will use following equation:
   $$d = \frac{1}{2} \cdot g \cdot t^2$$
   where:
- `d` is the distance fallen in **meters**,
- `g` is the standard acceleration of gravity in **meters per second squared**,
- `t` is the time spent falling in **seconds**.

In [7]:
def distance_falling(gravity, time):
    distance = 0.5 * gravity * time**2
    print(f"After {time} seconds, the object has fallen {distance:.2f} meters under a constant acceleration of {gravity} m/s².")
    return distance

The second function calculates the average speed of a falling object over a given time.

For this we will use following equation:
   $$v_{avg} = \frac{d}{t}$$
   where:
   - `v_avg` is the average speed in **meters per second**,
   - `d` is the distance fallen in **meters**,
   - `t` is the time spent falling in **seconds**.

In [2]:
def average_speed(distance, time):
    avg_speed = distance / time
    print(f"The average speed of an object that fell {distance:.2f} meters in {time} seconds is {avg_speed:.2f} m/s.")
    print()
    return avg_speed

***

### Task 2
The goal of this task is to use the above functions to compute the average speed of a falling object under different gravitational accelerations and for different falling times.

The gravitational accelerations used are as follows:
- Earth: $g$ = 9.80665 $m/s$ $^{[1]}$
- Moon: $g$ = 1.62468 $m/s$ $^{[2]}$
- Mars: $g$ = 3.70875 $m/s$ $^{[3]}$

The average speed will be calculated for an object that falls for the following durations:
- 10.0 seconds
- 20.0 seconds
- 30.0 seconds
- 40.0 seconds
- 50.0 seconds

In [15]:
earth_g = 9.80665
moon_g = 1.62468
mars_g = 3.70875

durations = [10.0, 20.0, 30.0, 40.0, 50.0]

#### Earth gravity

In [16]:
for times in durations:
    distance = distance_falling(earth_g, times)
    average_speed(distance, times)

After 10.0 seconds, the object has fallen 490.33 meters under a constant acceleration of 9.80665 m/s².
The average speed of an object that fell 490.33 meters in 10.0 seconds is 49.03 m/s.

After 20.0 seconds, the object has fallen 1961.33 meters under a constant acceleration of 9.80665 m/s².
The average speed of an object that fell 1961.33 meters in 20.0 seconds is 98.07 m/s.

After 30.0 seconds, the object has fallen 4412.99 meters under a constant acceleration of 9.80665 m/s².
The average speed of an object that fell 4412.99 meters in 30.0 seconds is 147.10 m/s.

After 40.0 seconds, the object has fallen 7845.32 meters under a constant acceleration of 9.80665 m/s².
The average speed of an object that fell 7845.32 meters in 40.0 seconds is 196.13 m/s.

After 50.0 seconds, the object has fallen 12258.31 meters under a constant acceleration of 9.80665 m/s².
The average speed of an object that fell 12258.31 meters in 50.0 seconds is 245.17 m/s.



#### Moon gravity

In [17]:
for times in durations:
    distance = distance_falling(moon_g, times)
    average_speed(distance, times)

After 10.0 seconds, the object has fallen 81.23 meters under a constant acceleration of 1.62468 m/s².
The average speed of an object that fell 81.23 meters in 10.0 seconds is 8.12 m/s.

After 20.0 seconds, the object has fallen 324.94 meters under a constant acceleration of 1.62468 m/s².
The average speed of an object that fell 324.94 meters in 20.0 seconds is 16.25 m/s.

After 30.0 seconds, the object has fallen 731.11 meters under a constant acceleration of 1.62468 m/s².
The average speed of an object that fell 731.11 meters in 30.0 seconds is 24.37 m/s.

After 40.0 seconds, the object has fallen 1299.74 meters under a constant acceleration of 1.62468 m/s².
The average speed of an object that fell 1299.74 meters in 40.0 seconds is 32.49 m/s.

After 50.0 seconds, the object has fallen 2030.85 meters under a constant acceleration of 1.62468 m/s².
The average speed of an object that fell 2030.85 meters in 50.0 seconds is 40.62 m/s.



#### Mars gravity

In [18]:
for times in durations:
    distance = distance_falling(mars_g, times)
    average_speed(distance, times)

After 10.0 seconds, the object has fallen 185.44 meters under a constant acceleration of 3.70875 m/s².
The average speed of an object that fell 185.44 meters in 10.0 seconds is 18.54 m/s.

After 20.0 seconds, the object has fallen 741.75 meters under a constant acceleration of 3.70875 m/s².
The average speed of an object that fell 741.75 meters in 20.0 seconds is 37.09 m/s.

After 30.0 seconds, the object has fallen 1668.94 meters under a constant acceleration of 3.70875 m/s².
The average speed of an object that fell 1668.94 meters in 30.0 seconds is 55.63 m/s.

After 40.0 seconds, the object has fallen 2967.00 meters under a constant acceleration of 3.70875 m/s².
The average speed of an object that fell 2967.00 meters in 40.0 seconds is 74.17 m/s.

After 50.0 seconds, the object has fallen 4635.94 meters under a constant acceleration of 3.70875 m/s².
The average speed of an object that fell 4635.94 meters in 50.0 seconds is 92.72 m/s.



***

#### Summary of Results and Interpretation

The results show that the falling distance and the average speed both increase as the falling time increases.

The falling distance increases quadratically with time. This means that when the falling time doubles, the distance fallen becomes four times larger.

The average speed increases linearly with time. Resulting in the following: stronger gravitational acceleration leads to a larger average speed for the same falling time.

Comparing the three gravitational accelerations, Earth produces the largest falling distances and average speeds, the Moon produces the smallest values, and Mars lies between them. 
After 50 seconds, the object falls about 12258.31 meters on Earth, about 4635.94 meters on Mars, and about 2030.85 meters on the Moon.

These differences stems from the fact that the gravitational acceleration is strongest on Earth, weaker on Mars, and weakest on the Moon.


***

## References

[1] Boynton, R. (2001). Precise measurement of mass. 60th Annual Conference of the Society of Allied Weight Engineers, Paper, Arlington, Texas May 21-23, 2001.

[2] Hirt, C., and Featherstone, W. E. (2012). A 1.5 km-resolution gravity field model of the Moon. Earth and Planetary Science Letters, 329, 22-30.

[3] Hirt, C., Claessens, S. J., Kuhn, M., and Featherstone, W. E. (2012). Kilometer-resolution gravity field of Mars: MGM2011. Planetary and Space Science, 67(1), 147-154.